python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_condition.py

```diff
@@ -0,0 +1,61 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Optional, Sequence
+
+from dagster._core.definitions.declarative_scheduling.serialized_objects import (
+    AssetConditionSnapshot,
+)
+from dagster._model import DagsterModel
+from dagster._utils.security import non_secure_md5_hash_str
+
+if TYPE_CHECKING:
+    from .legacy.asset_condition import AssetConditionResult
+    from .scheduling_context import SchedulingContext
+
+
+class SchedulingCondition(ABC, DagsterModel):
```

Comment:
Might consider naming the `ABC` `ISchedulingCondition` or something because I will assume you will want a concrete `SchedulingCondition` later.